Automattic/wp-calypso
# Store: Settings: Prompt for save if settings have been edited

Fixes #17271

Design

At first I was going to do something elaborate where I tested the queued changes. That got messy and brittle. So I changed to this approach where starting an edit raises a flag and saving clears it.

To test:

Ensure that if you attempt to navigate away from making settings changes, you are prompted to confirm (or cancel) navigation before it happens. Try editing settings like

* Payments > Store Currency
* Payments > { Payment Method } > Enabled
* Payments > Direct bank transfer} > Manage > Title
* Shipping > Add Package
* Shipping > { Package } > Edit
* Taxes > Tax Calculations Enabled
* Taxes > Tax settings > Charge taxes on shipping costs
* Email > MailChimp > Newsletter subscriptions > Show a subscribe message to customer at checkout
